Remove twitter Quirks related to web-share

karlcow opened the pull request:
#### 29b13de20953a2bdfdb1f5dfa7bf3c856d120009
<pre>
Remove twitter Quirks related to web-share
<a href="https://bugs.webkit.org/show_bug.cgi?id=243088">https://bugs.webkit.org/show_bug.cgi?id=243088</a>

Reviewed by Youenn Fablet.

WebKit is now using the &quot;*&quot; permissions policy
for &quot;web-share&quot; and we are coordinating with
Twitter to add &quot;web-share&quot; to their
video-embedding iframes.

YouTube is also being contacted to add &quot;web-share&quot;.

We can remove the Quirks and related functions.

Manually tested.

* Source/WebCore/page/Navigator.cpp:
(WebCore::validateWebSharePolicy):
* Source/WebCore/page/Quirks.cpp:
(WebCore::isTwitterDocument): Deleted.
(WebCore::isYouTubeDocument): Deleted.
(WebCore::Quirks::requiresUserGestureToLoadInPictureInPicture const):
(WebCore::Quirks::shouldDisableWebSharePolicy const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: <a href="https://commits.webkit.org/252866@main">https://commits.webkit.org/252866@main</a>
</pre>

youennf reviewed (approved):
Do we even need isYouTubeDocument anymore given we are using '*' as a default?
We might be able to remove shouldDisableWebSharePolicy()

karlcow committed:
Remove twitter Quirks related to web-share
https://bugs.webkit.org/show_bug.cgi?id=243088

Reviewed by Youenn Fablet.

WebKit is now using the "*" permissions policy
for "web-share" and we are coordinating with
Twitter to add "web-share" to their
video-embedding iframes.

YouTube is also being contacted to add "web-share".

We can remove the Quirks and related functions.

Manually tested.

* Source/WebCore/page/Navigator.cpp:
(WebCore::validateWebSharePolicy):
* Source/WebCore/page/Quirks.cpp:
(WebCore::isTwitterDocument): Deleted.
(WebCore::isYouTubeDocument): Deleted.
(WebCore::Quirks::requiresUserGestureToLoadInPictureInPicture const):
(WebCore::Quirks::shouldDisableWebSharePolicy const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: https://commits.webkit.org/252866@main
